groups/ecs-service/main.tf

```diff
@@ -24,7 +24,7 @@ module "ecs-service" {
 
   # Docker container details
   docker_registry   = var.docker_registry
-  docker_repo       = local.docker_repo
+  docker_repo       = var.docker_repo != "" ? var.docker_repo : local.docker_repo
```

Comment:
As we're already referencing a local variable here, it would be best to have this logic defined where the local itself is created in `locals.tf`